docs/code-style-guide.md

```diff
@@ -0,0 +1,211 @@
+
+If you want to know more about our coding style you can take a look at our [clang-format configuration file](https://github.com/UBC-Thunderbots/Software/blob/master/.clang-format).
+
+{% hint style="info" %}
+Each coding style has a tag \(i.e. _N1_\), use this tag to comment on code reviews.
+{% endhint %}
+
+### Names and Variables
+
+* Classes, structures, namespaces, unions, enumerates, "typename"-type template parameters, and typedefs names uses `CamelCase` with a capital letter.
+
+  ```cpp
+  // Incorrect
+  class eventHandler;
+
+  // Incorrect
+  class event_handler;
+
+  // Correct
+  class EventHandler;
+  ```
+
+* Constant variables uses `ALL_CAPS_WITH_UNDERSCORES`. Constant variables include _static const_, _const class members_, _const_ global variables, and _const_ enumerations.
+* All variable names are `lowercase_with_underscores`
+  ```cpp
+  // Incorrect
+  float calculatedDistnace;
+
+  // Correct
+  float calculated_distance;
+  ```
+* Function and method names are `camelCase` with leader lowercase letter.
+  ```cpp
+  // Incorrect
+  float get_distance();
+
+  // Incorrect
+  float GetDistance();
+
+  // Correct
+  float getDistance();
+  ```
+* File names are lower case and can contain underscores. The name should match or similar to the class/classes/functions it contains. Avoid using ambiguous names or names already exist in common libraries.
+  ```cpp
+  // Incorrect
+  ShootGoal.h
+
+  // Incorrect
+  sg.h
+
+  // Acceptable
+  shootgoal.h
+
+  // Correct
+  shoot_goal.h
+  ```
+* Avoid "obvious" or "magic" numbers unless it's part of a mathematical or physics formula (ex `A=0.5(b*h)`).
+  ```cpp
+  // Incorrect
+  float distance = catch_distance * 2.15;
+
+  // Correct
+  const float SCALE_FACTOR = 2.15;
+  float distance = catch_distance * SCALE_FACTOR;
+  ```
+
+### Comments
+
+Make sure to comment both the interface for a function or class, as well as the logic in the implementation. In general, try to make as many in-code documentations whenever possible.
+
+As much code documentation as possible should live with the code itself \(in the form of comments\). This makes it easier for people working on the code to find the information, and because the code and comments are version-controlled together if we ever go back to an older version of the code, we will have the corresponding older documentation as well.
+
+*Code comments are very important. Be sure to comment your code well enough so that another member of the team would be able to quickly get an understanding of what your code is doing, **and why**. Try not to make your comments too unnecessarily verbose, but include as much detail you feel is necessary to get the necessary points across. We realize that sounds contradictory, but use your best judgement as to what you think is clear and readable.*
```

Comment:
UBER nit warning 2:

There are a lot of 'neccisarries' going on here.

I would rephrase to 'but include as much detail you feel is required to adequately explain the code'

Reply:
changed